src/sisl/_core/atom.py

```diff
@@ -758,9 +758,40 @@ def Z(self):
         uZ = _a.arrayi([a.Z for a in self.atom])
         return uZ[self.species]
 
-    def index(self, atom):
-        """Return the indices of the atom object"""
-        return (self._species == self.species_index(atom)).nonzero()[0]
+    def index(self, *atoms):
```

Comment:
Having seen this interface, I think it's better to have `atoms` as a single argument. In this way, we can enable keyword arguments later (well, your current API does not limit that, but it's very inconsistent with the rest of sisl).

Instead it should just be:
```python
def index(self, atoms)
```